### PR TITLE
Added 'total downloaded' from server and peers in the player

### PR DESCRIPTION
**Pull request by Poslovitch**
## Description

This PR implements #3384 with the changes @Chocobozzz suggested in https://github.com/Chocobozzz/PeerTube/issues/3384#issuecomment-737112057. Namely:
* adding a `source` field in `PlayerNetworkInfo`
* adding the requested information in `subDivWebtorrent.title` (from `p2p-info-button.ts`) **if** the server uses HLS.

WebTorrent is seemingly not able to tell the difference between the server and other peers, hence why it's left out of that PR.

## Related issues

Resolves https://github.com/Chocobozzz/PeerTube/issues/3384

## Has this been tested?

- [ ] 👍 yes, I added tests to the test suite
- [ ] 👍 yes, light tests as follows are enough
- [ ] 💭 no, because this PR is a draft and still needs work
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Screenshots

image

### CODE REVIEW

File: client/src/assets/player/videojs-components/p2p-info-button.ts [hunk 1]
```diff
@@ -84,8 +84,15 @@ class P2pInfoButton extends Button {
...
-        this.player().localize('Total uploaded: ' + totalUploaded.join(' '))
+      subDivWebtorrent.title = this.player().localize('Total downloaded: ') + totalDownloaded.join(' ') + '\n'
+      if (data.source == 'p2p-media-loader') {
+        const downloadedFromServer = bytes(httpStats.downloaded).join(' ')
+        const downloadedFromPeers = bytes(p2pStats.downloaded).join(' ')
+
+        subDivWebtorrent.title += '* ' + this.player().localize('From the server: ') + downloadedFromServer + '\n' +
+          '* ' + this.player().localize('From peers: ') + downloadedFromPeers + '\n'
+      }
+      subDivWebtorrent.title += this.player().localize('Total uploaded: ' + totalUploaded.join(' '))
```
Comment (rigelk) [inline client/src/assets/player/videojs-components/p2p-info-button.ts:95]:
```suggestion
      subDivWebtorrent.title += this.player().localize('Total uploaded: ') + totalUploaded.join(' ')
```

File: client/src/assets/player/videojs-components/p2p-info-button.ts [hunk 2]
```diff
@@ -84,8 +84,15 @@ class P2pInfoButton extends Button {
...
 
-      subDivWebtorrent.title = this.player().localize('Total downloaded: ') + totalDownloaded.join(' ') + '\n' +
-        this.player().localize('Total uploaded: ' + totalUploaded.join(' '))
+      subDivWebtorrent.title = this.player().localize('Total downloaded: ') + totalDownloaded.join(' ') + '\n'
+      if (data.source == 'p2p-media-loader') {
+        const downloadedFromServer = bytes(httpStats.downloaded).join(' ')
+        const downloadedFromPeers = bytes(p2pStats.downloaded).join(' ')
+
+        subDivWebtorrent.title += '* ' + this.player().localize('From the server: ') + downloadedFromServer + '\n' +
+          '* ' + this.player().localize('From peers: ') + downloadedFromPeers + '\n'
```
Comment (rigelk) [inline client/src/assets/player/videojs-components/p2p-info-button.ts:93]:
```suggestion
        subDivWebtorrent.title +=
          '* ' + this.player().localize('From the server: ') + downloadedFromServer + '\n' +
          '* ' + this.player().localize('From peers: ') + downloadedFromPeers + '\n'
```

File: client/src/assets/player/videojs-components/p2p-info-button.ts [hunk 3]
```diff
@@ -84,8 +84,15 @@ class P2pInfoButton extends Button {
       const totalUploaded = bytes(p2pStats.uploaded + httpStats.uploaded)
       const numPeers = p2pStats.numPeers
 
-      subDivWebtorrent.title = this.player().localize('Total downloaded: ') + totalDownloaded.join(' ') + '\n' +
-        this.player().localize('Total uploaded: ' + totalUploaded.join(' '))
+      subDivWebtorrent.title = this.player().localize('Total downloaded: ') + totalDownloaded.join(' ') + '\n'
+      if (data.source == 'p2p-media-loader') {
```
Comment (rigelk) [inline client/src/assets/player/videojs-components/p2p-info-button.ts:88]:
```suggestion

      if (data.source === 'p2p-media-loader') {
```

**Review by rigelk — CHANGES_REQUESTED** [review]
Aside from some quirks, LGTM! Thanks for your contribution @Poslovitch!

**Review by rigelk — APPROVED** [review]

**Comment by Booteille** [comment]
Thanks everyone!

**Pull request merged**